### PR TITLE
[Feature] Add OneCycleLR and CyclicLR scheduler

### PR DESCRIPTION
**Pull request by The-truthh**
Thank you for your contribution to the MindCV repo.
Before submitting this PR, please make sure:

- [ ] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved terminology
- [ ] You have added unit tests

## Motivation

Add OneCycleLR and CyclicLR scheduler

## Test Plan

It's tested in dynamic_lr.

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: mindcv/scheduler/dynamic_lr.py [hunk 1]
```diff
@@ -191,6 +191,178 @@ def cosine_annealing_warm_restarts_lr(te, tm, eta_min, *, eta_max, steps_per_epo
     return lrs
 
 
+def one_cycle_lr(pct_start, anneal_strategy, three_phase, min_lr, max_lr, *, lr, steps_per_epoch, epochs):
```
Comment (SamitHuang) [inline mindcv/scheduler/dynamic_lr.py:194]: 建议加注释注明方法出处

File: train.py [hunk 2]
```diff
@@ -171,15 +171,22 @@ def train(args):
...
         num_epochs=args.epoch_size,
-        lr_epoch_stair=args.lr_epoch_stair,
         num_cycles=args.num_cycles,
         cycle_decay=args.cycle_decay,
+        pct_step_up=args.pct_step_up,
+        decay_strategy=args.decay_strategy,
+        three_phase=args.three_phase,
+        step_size_up=args.step_size_up,
+        step_size_down=args.step_size_down,
+        decay_mode=args.decay_mode,
```
Comment (SamitHuang) [inline train.py:188]: 这堆decay参数过于confusing. 建议优化命名，或者加个前缀区分下。考虑精简下参数个数。如：
decay_strategy -> oc_decay_strategy
three_phase -> is_oc_three_phase
step_size_up -> oc_ascend_steps
step_size_down -> oc_descend_steps

File: config.py [hunk 3]
```diff
@@ -182,12 +182,15 @@ def create_parser():
...
     group.add_argument('--scheduler', type=str, default='cosine_decay',
-                       choices=['constant', 'cosine_decay', 'exponential_decay', 'step_decay', 'multi_step_decay'],
+                       choices=['constant', 'cosine_decay', 'exponential_decay', 'step_decay',
+                                'multi_step_decay', 'one_cycle', 'cyclic'],
                        help='Type of scheduler (default="cosine_decay")')
     group.add_argument('--lr', type=float, default=0.001,
                        help='Learning rate (default=0.001)')
     group.add_argument('--min_lr', type=float, default=1e-6,
                        help='The minimum value of learning rate if scheduler supports (default=1e-6)')
+    group.add_argument('--max_lr', type=float, default=2.0,
```
Comment (geniuspatrick) [inline config.py:192]: 这个是不是没有必要加，lr就能代表max_lr

File: mindcv/scheduler/dynamic_lr.py [hunk 4]
```diff
@@ -191,6 +191,178 @@ def cosine_annealing_warm_restarts_lr(te, tm, eta_min, *, eta_max, steps_per_epo
...
+                    lrs.append(anneal_func(lr, max_lr, i / step_size_up))
+                else:
+                    lrs.append(anneal_func(max_lr, min_lr, (i - step_size_up) / (step_size_end - step_size_up)))
+            lr_epoch = lrs[-1]
+        else:
+            lrs.append(lr_epoch)
+    return lrs
+
+
+def one_cycle_refined_lr(pct_start, anneal_strategy, three_phase, min_lr, max_lr, *, lr, steps_per_epoch, epochs):
```
Comment (geniuspatrick) [inline mindcv/scheduler/dynamic_lr.py:233]: 这个应该没有refined吧，一直都是随step变化的

Comment (geniuspatrick) [reply mindcv/scheduler/dynamic_lr.py:233]: momentum是不是没法做

File: train.py (hunk 2 above)
Comment (geniuspatrick) [reply train.py:188]: 这个可能先不用暴露出来吧，感觉太过复杂。在create_scheduler这里不传入这些参数

File: config.py (hunk 3 above)
Comment (The-truthh) [reply config.py:192]: 不行，根据论文中的设置，resnet one cycle的学习率变化为0.05（lr）到1.0（max_lr），最后下降至0.00005（min_lr）

Comment (The-truthh) [reply config.py:192]: 而且这样也无法使用warm_up 来达到初始学习率了

File: mindcv/scheduler/scheduler_factory.py [hunk 5]
```diff
@@ -90,6 +94,41 @@ def create_scheduler(
...
+        if lr_epoch_stair or warmup_epochs > 0:
+            raise ValueError(
+                "OneCycle scheduler doesn't support learning rate varies with epoch and warmup_epochs > 0."
+            )
+        div_factor = 25.0
+        initial_lr = lr / div_factor
+        final_div_factor = initial_lr / min_lr
+        main_lr_scheduler = one_cycle_lr(
+            max_lr=lr,
+            pct_start=0.3,
```
Comment (geniuspatrick) [inline mindcv/scheduler/scheduler_factory.py:107]: 这些默认值的，不用写在factory里就行

**Review by geniuspatrick — CHANGES_REQUESTED** [review]

File: mindcv/scheduler/scheduler_factory.py (hunk 5 above)
Comment (The-truthh) [reply mindcv/scheduler/scheduler_factory.py]: 已修改

**Review by geniuspatrick — APPROVED** [review]

**Review by SamitHuang — APPROVED** [review]
